src/logic/signupLogic.ts

```diff
@@ -0,0 +1,118 @@
+import { isValidEmailAddress } from 'lib/utils'
+import { posthogAnalyticsLogic } from './posthogAnalyticsLogic'
+
+export enum SignupModalView {
+    EMAIL_PROMPT = 'EMAIL_PROMPT',
+    DEPLOYMENT_OPTIONS = 'DEPLOYMENT_OPTIONS',
+}
+
+export enum Realm {
+    hosted = 'hosted',
+    cloud = 'cloud',
+}
+
+type HubSpotContact = {
+    id: string // Hubspot contact ID (number as string)
+    properties: Record<string, string>
+    created_at: string // ISO 8601
+    updated_at: string // ISO 8601
+    archived: boolean
+    archived_at: string | null // ISO 8601
+}
+
+export type HubSpotContactResponse = {
+    success: boolean
+    result: string | HubSpotContact
+}
+
+async function createContact(email: string) {
+    const url = process.env.GATSBY_POSTHOG_API_HOST + '/create_web_contact'
+    const body = new FormData()
+    body.append('email', email)
+    return (
+        await fetch(url, {
+            method: 'POST',
+            body,
+        })
+    ).json() as Promise<HubSpotContactResponse>
+}
+
+async function updateContact(email: string, properties: Record<string, string>) {
+    const url = process.env.GATSBY_POSTHOG_API_HOST + '/update_web_contact'
+    const body = new FormData()
+    body.append('email', email)
+    for (const [property, value] of Object.entries(properties)) {
+        body.append(property, value)
+    }
+    return (
+        await fetch(url, {
+            method: 'POST',
+            body,
+        })
+    ).json() as Promise<HubSpotContactResponse>
+}
+
+export const signupLogic = kea({
+    actions: {
+        setModalView: (view: SignupModalView) => ({ view }),
+        setEmail: (email: string) => ({ email }),
+        submitForm: true,
+        skipEmailEntry: true,
+        reportModalShown: true,
+        reportDeploymentTypeSelected: (deploymentType: Realm, nextHref?: string) => ({ deploymentType, nextHref }),
+    },
+    reducers: {
+        modalView: [
+            SignupModalView.EMAIL_PROMPT as SignupModalView,
+            {
+                setModalView: (_, { view }: { view: SignupModalView }) => view,
+            },
+        ],
+        email: [
+            '',
+            {
+                setEmail: (_, { email }: { email: string }) => email,
+            },
+        ],
+    },
+    connect: {
+        values: [posthogAnalyticsLogic, ['posthog']],
+    },
+    listeners: ({ actions, values }) => ({
+        submitForm: async () => {
+            const { posthog, email } = values
+            if (email && isValidEmailAddress(email)) {
+                try {
+                    posthog?.identify(email, { email }) // use email as distinct ID; also set it as property
+                    posthog?.capture('signup: submit email')
+                    await createContact(email)
+                    actions.setModalView(SignupModalView.DEPLOYMENT_OPTIONS)
+                } catch (err) {
+                    posthog?.capture('signup: failed to create contact', { message: err })
```

Comment:
Do we suspect there's a high chance posthog isn't initiated here? 

If so, could we log that to Sentry? Would definitely need that logging if we expect a potential data gap between what we're seeing in PostHog versus what's actually going into HubSpot.

Reply:
Not sure. `posthog` starts out as null, and is then initialized once the posthog-js script is loaded on the page. Though, I assume this will all happen before signupLogic is mounted. I'll just remove the `?` guards, and if PH turns out not to be initialized, we'll see errors come in and know what to fix.

I think Kea is also slightly out of date on this repo — `posthog` should be typed auto-magically.